ab_tool/static/ab_tool/js/experiments_dashboard.js

```diff
@@ -21,73 +67,53 @@ $(document).ready(function(){
-            
-            //as they change the input url field
-            $(this).on('input', function(){
-                if ( is_valid_url( $(this).val().trim() ) ){
-                    
-                    if ( $(this).siblings('.preview-link').is(":hidden") ){
-                        $(this).siblings('.preview-link').show();
-                    }
-
-                    $(this)
-                        .siblings('.preview-link')
-                        .attr({
-                            href:$(this).val().trim(),
-                            target: '_blank'
-                        });
-                }else{
-                    $(this).siblings('.preview-link').hide();
-                }
-                
-            });
-
-        });//end each
-
-        //e.currentTarget = references the modal window that's open
-        var $currentModal = $(e.currentTarget);
-        //find the intervention name for the window that's open
-        var $interventionName = $currentModal.find('.intervention-name');
-
-        //disable the create button if IP name is empty
-        //this is mostly for new IP
-        if ( !$.trim( $interventionName.val() ).length ){
-            $currentModal.find('button').addClass('disabled');
+
+    $('[data-modal="intervention-new"]').each(function() {
+        reset_IP_create_form($(this).find('.modal-form'));
+    });
+
+    $('[data-modal="intervention-edit"]').each(function() {
+        reset_IP_edit_form($(this).find('.modal-form'));
+    });
+
+    // Set up event handlers ****************************************************************************************
+
+    // multiple-submission protection for create/update button
+    $('.modal-form').submit(function (event) {
+        var $submit = $(this).find('.modal-submit');
+        if ($submit.hasClass("disabled")) {
+            event.preventDefault();
         }
-        
-        //update error class and able/disabled button as the user
-        //changes the intervention name input
-        $interventionName.on('input', function(){
-            if ( $.trim( $(this).val() ).length ){
-                $currentModal.find('button').removeClass('disabled');
-                if ( $(this).hasClass('has-error') ){
-                    $(this).removeClass('has-error');
-                }
-            }else{
-                $currentModal.find('button').addClass('disabled');
-                $(this).addClass('has-error');
-            }
-        });
+        else {
+            // disables create/update button, cancel button, and marks form for submit
+            // (so that subsequent calls to enable_submit_if_no_errors will not re-enable form)
+            $submit.addClass("disabled submitting");
+            $submit.siblings('.modal-cancel').addClass("disabled");
+        }
+    });
+
+    $('.intervention-url').on('input', function(){
+        validate_url_input_element($(this));
+        enable_submit_if_no_errors($(this).closest('.modal-form'));
```

Comment:
Not that you should fix all of these now, but we should stick to camel-case in Javascript.